cmd/cdi-cloner/clone-source.go

```diff
@@ -193,23 +196,24 @@ func newTarReader(preallocation bool) (io.ReadCloser, error) {
 	return &execReader{cmd: cmd, stdout: stdout, stderr: io.NopCloser(&stderr)}, nil
 }
 
-func getInputStream(preallocation bool) (rc io.ReadCloser) {
-	var err error
+func getInputStream(preallocation bool) io.ReadCloser {
 	switch contentType {
 	case "filesystem-clone":
-		rc, err = newTarReader(preallocation)
+		rc, err := newTarReader(preallocation)
 		if err != nil {
 			klog.Fatalf("Error creating tar reader for %q: %+v", mountPoint, err)
 		}
+		return rc
 	case "blockdevice-clone":
-		rc, err = os.Open(mountPoint)
+		rc, err := os.Open(mountPoint)
 		if err != nil {
 			klog.Fatalf("Error opening block device %q: %+v", mountPoint, err)
 		}
+		return rc
 	default:
 		klog.Fatalf("Invalid content-type %q", contentType)
+		return nil
```

Comment:
Nit: Not a must but I'd leave this return outside the switch, imo it's better always to have an unnested final return.